### ISSUE TITLE
vcf: Constrain CHROM record field with the same contig header record ID rules

### ISSUE THREAD

**Issue by zaeleus** [samtools/hts-specs#712]
The ID for a contig header record is constrained by the following regular expression (§ 1.4.7 "Contig field format") with additional rules (#711) omitted here:

> `[0-9A-Za-z!#$%&+./:;?@^_|~-][0-9A-Za-z!#$%&*+./:;=?@^_|~-]*`

However, there is a mismatch for the CHROM record field, where the description of constraints are "String, no whitespace permitted, Required" (§ 1.6.1.1 "CHROM"), i.e., `^[^\s]*$`. This clearly allows far more than what is for a valid contig name.

The CHROM record field should match the same rules as the contig header record ID.

**Comment by jmarshall** [comment]
Indeed. Previously noted as the second bullet point that “the VCF maintainers may wish to consider” in PR #379, but sadly that was not addressed.
